src/dmd/mars.d

```diff
@@ -1406,20 +1397,56 @@ void addDefaultVersionIdentifiers()
     VersionCondition.addPredefinedGlobalIdent("D_HardFloat");
 }
 
-private void printPredefinedVersions()
+private void printPredefinedVersions(FILE* stream)
 {
-    if (global.params.verbose && global.versionids)
+    if (global.versionids)
     {
         OutBuffer buf;
         foreach (const str; *global.versionids)
         {
             buf.writeByte(' ');
             buf.writestring(str.toChars());
         }
-        message("predefs  %s", buf.peekString());
+        stream.fprintf("predefs  %.*s", buf.peekString());
     }
 }
 
+extern(C) void printGlobalConfigs(FILE* stream)
+{
+    stream.fprintf("binary    %s\n", global.params.argv0);
+    stream.fprintf("version   %s\n", global._version);
+    stream.fprintf("config    %s\n", global.inifilename ? global.inifilename : "(none)");
+    // Print DFLAGS environment variable
+    {
+        StringTable environment;
+        environment._init(0);
+        Strings dflags;
+        getenv_setargv(readFromEnv(&environment, "DFLAGS"), &dflags);
+        environment.reset(1);
+        OutBuffer buf;
+        foreach (flag; dflags.asDArray)
+        {
+            bool needsQuoting;
+            for (auto flagp = flag; flagp; flagp++)
```

Comment:
Wat? Since when does incrementing a pointer until it is null work?

Reply:
For null terminated strings?

Reply:
FWIW I just _moved_ the code - check the red part on the left.